### PR TITLE
infra(server): fix Album TypeORM relations and change ids to uuids

### PR DESCRIPTION
**Pull request by zackpollard**
Updates Album TypeORM entity to use correct ManyToMany relations for sharedUsers and assets.
Updates API Key TypeORM entity to use UUIDs for the ID column.

Part 1 of fixing TypeORM entities to follow the same standards

### CODE REVIEW

**Comment by jrasm91** [comment]
Worked on this a bit here: https://github.com/immich-app/immich/tree/infra/fix-relations-2

- Added back the migration (that's been released now) with the album owner FK (drop and re-add in the new migration since the cascade is a little bit different)
- Reloaded the album after calling `save` (using `({ id: * } as *Entity` means the relation won't be fully loaded otherwise)
- Removed `DataSource` dependency
- Finished the last `getSharedWithUserAlbumCount` count implementation
- Rebased onto main

Did some basic testing - was able to create an album, a shared album, add users, remove users, create a shared link for an album, etc. Seems to be working.

Didn't do anything with tests :laughing:

File: server/libs/domain/src/share/response-dto/shared-link-response.dto.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class SharedLinkResponseDto {
 
 export function mapSharedLink(sharedLink: SharedLinkEntity): SharedLinkResponseDto {
   const linkAssets = sharedLink.assets || [];
-  const albumAssets = (sharedLink?.album?.assets || []).map((albumAsset) => albumAsset.assetInfo);
+  const albumAssets = (sharedLink?.album?.assets || []).map((asset) => asset);
```
Comment (jrasm91) [inline server/libs/domain/src/share/response-dto/shared-link-response.dto.ts:26]: Probably can delete the map function here

File: server/libs/domain/src/share/response-dto/shared-link-response.dto.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export function mapSharedLink(sharedLink: SharedLinkEntity): SharedLinkResponseD
 
 export function mapSharedLinkWithNoExif(sharedLink: SharedLinkEntity): SharedLinkResponseDto {
   const linkAssets = sharedLink.assets || [];
-  const albumAssets = (sharedLink?.album?.assets || []).map((albumAsset) => albumAsset.assetInfo);
+  const albumAssets = (sharedLink?.album?.assets || []).map((asset) => asset);
```
Comment (jrasm91) [inline server/libs/domain/src/share/response-dto/shared-link-response.dto.ts:48]: And here

File: server/libs/infra/src/db/entities/album.entity.ts [hunk 3]
```diff
@@ -35,11 +36,13 @@ export class AlbumEntity {
...
-  sharedUsers?: UserAlbumEntity[];
+  @ManyToMany(() => UserEntity, { eager: true })
+  @JoinTable()
+  sharedUsers!: UserEntity[];
 
-  @OneToMany(() => AssetAlbumEntity, (assetAlbumEntity) => assetAlbumEntity.albumInfo)
-  assets?: AssetAlbumEntity[];
+  @ManyToMany(() => AssetEntity, { eager: true })
+  @JoinTable()
+  assets!: AssetEntity[];
```
Comment (jrasm91) [inline server/libs/infra/src/db/entities/album.entity.ts:45]: This is current behavior, so it's fine, but might have to re-visit it. Not sure what performance impact is for loading all of the assets for all of the albums from the database on a large (album-heavy) library.

**Review by jrasm91 — APPROVED** [review]

**Comment by alextran1502** [comment]
Tested on functionality. No problem with the migration of existing data from `main`

- [x] create and access owned album
- [x] create and access internal share album
- [x] create and access public shared album

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**